### PR TITLE
[REVIEW] Adding macros: GDF_TRY(), CUDF_TRY(), ASSERT_CUDF_SUCCEEDED()

### PR DESCRIPTION
**Pull request by eyalroz**
See issue #1369 . This was split off from PR #1366.

### CODE REVIEW

File: cpp/src/utilities/error_utils.hpp [hunk 1]
```diff
@@ -25,6 +25,19 @@
...
   if (!(F)) return (S);
 
+/**---------------------------------------------------------------------------*
+ * @brief a version of GDF_REQUIRE for expressions of type `gdf_error` rather
+ * than booleans
+ *
+ * This macro is sort-of DEPRECATED.
+ *
+ *---------------------------------------------------------------------------**/
+#define GDF_TRY(...) do { \
```
Comment (jrhemstad) [inline cpp/src/utilities/error_utils.hpp]: Can you give me an example of why this need to be variadic? I would think it should be the same as `RMM_TRY` above and just accept a single parameter.

Comment (eyalroz) [reply cpp/src/utilities/error_utils.hpp]: Actually, not really. Will make it a unary macro.

Comment (eyalroz) [reply cpp/src/utilities/error_utils.hpp]: Oh, and still need to add the other macro.

File: cpp/tests/utilities/cudf_test_fixtures.h [hunk 2]
```diff
@@ -26,6 +27,15 @@
 #define ASSERT_RMM_SUCCEEDED(expr)  ASSERT_EQ(RMM_SUCCESS, expr)
 #define EXPECT_RMM_SUCCEEDED(expr)  EXPECT_EQ(RMM_SUCCESS, expr)
 
+#define ASSERT_CUDF_SUCCEEDED(gdf_error_expression) \
```
Comment (jrhemstad) [inline cpp/tests/utilities/cudf_test_fixtures.h]: Should wrap in the standard `do {...} while(0)`

Comment (eyalroz) [reply cpp/tests/utilities/cudf_test_fixtures.h]: Are you sure? there's no if here.

Comment (eyalroz) [reply cpp/tests/utilities/cudf_test_fixtures.h]: Done.

Comment (eyalroz) [reply cpp/tests/utilities/cudf_test_fixtures.h]: @jrhemstad : Please approve so this can be merged.

**Pull request merged**